src/toil/wdl/wdl_analysis.py

```diff
@@ -21,6 +21,41 @@
 wdllogger = logging.getLogger(__name__)
 
 
+class WDLType(str):
+    """
+    Represents a primitive or compound WDL type:
+
+    https://github.com/openwdl/wdl/blob/main/versions/development/SPEC.md#types
+    """
```

Comment:
Can you explain a bit about why this has to extend `str`? I don't quite get it.

If we need to change our internal representation of WDL types over to actual classes to make `Pair` and `Map` work, why not change all of them and have `WDLIntType`, etc.? Then I think we wouldn't need to extend `str` here.

Reply:
Extending `str` is certainly not the best approach, but it requires the least amount of refactoring since the rest of the implementation expects type to be a `str`. 

My intention with this PR is not to refactor the entire internal representation of WDL types, since @DailyDreaming has suggested that we should probably rework the overall design of the WDL implementation. 

If this is something we would want to have for future implementations, then I can adjust my tickets and focus on this instead.